### PR TITLE
Migrate all minitest tests to rspec

### PR DESCRIPTION
**Pull request by TheWalkingLeek**

### CODE REVIEW

File: spec/integration/create_team_spec.rb [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
+describe 'Create team' do
+  include IntegrationHelpers::DefaultHelper
+  include IntegrationHelpers::AccountTeamSetupHelper
 
-  setup do
-    GeoIp.expects(:activated?).returns(false).at_least_once
-  end
-
-  test 'bob creates new normal team' do
+  it 'creates new normal team as bob' do
```
Comment (Thunderguts13) [inline spec/integration/create_team_spec.rb:13]: context 'as bob' do

File: spec/integration/create_user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        givenname: 'Simon',
+        surname: 'Kern'
+      } }
+      expect(response).to redirect_to admin_users_path
+      expect(User::Human.find_by(username: 'simon')).to_not be_nil
+      logout
+      login_as('simon')
+    end
+
+    it 'cannot create second user bob as admin' do
```
Comment (Thunderguts13) [inline spec/integration/create_user_spec.rb:28]: remove as admin it's already in the context

File: spec/integration/ldap_user_login_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    enable_ldap
+  end
+
+  it 'logins as ldap user' do
+    ldap = double
+
+    # Prepare for  do
+    user_bob = users(:bob)
+    user_bob.update(auth: 'ldap')
+    user_bob.update(ldap_uid: 42)
```
Comment (Thunderguts13) [inline spec/integration/ldap_user_login_spec.rb:23]: Move to before block if it has to be done in each test

File: spec/policies/account_policy_spec.rb [hunk 4]
```diff
@@ -1,129 +1,115 @@
...
-      assert_permit api_bob, account2, :show?
+  let(:account1) { accounts(:account1) }
+  let(:account2) { accounts(:account2) }
+  let(:group2) { groups(:group2) }
+  let(:team1) { teams(:team1) }
+  let(:team2) { teams(:team2) }
+
+  context 'as teammember' do
+    it 'can show account' do
+      assert_permit bob, account2, :show?
```
Comment (Thunderguts13) [inline spec/policies/account_policy_spec.rb:15]: I would prefer to have the expect in the rspec class. Like `expect(permited?(bob, account2, :show?).to be true`

File: spec/policies/user/human_policy_spec.rb [hunk 5]
```diff
@@ -1,205 +1,209 @@
...
+      it 'can destroy a ldap user' do
         #  LdapConnection.any_instance.expects(:exists?)
         #                             .with('bob')
         #                             .returns(true)
 
         assert_permit admin, ldap_user, :destroy?
       end
 
-      # test 'admin can destroy a removed ldap user' do
+      # it 'admin can destroy a removed ldap user' do
```
Comment (Thunderguts13) [inline spec/policies/user/human_policy_spec.rb:206]: either fix the test or remove it

File: spec/policies/user/human_policy_spec.rb [hunk 6]
```diff
@@ -1,205 +1,209 @@
...
         assert_permit admin, conf_admin, :destroy?
       end
 
-      test 'admin can destroy a user' do
+      it 'can destroy a user' do
         assert_permit admin, bob, :destroy?
       end
 
-      test 'admin can destroy a ldap user' do
+      it 'can destroy a ldap user' do
```
Comment (Thunderguts13) [inline spec/policies/user/human_policy_spec.rb:198]: remove commented code

File: spec/unit/lib/ldap_connection_spec.rb [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+      # bind by bind_dn
+      expect_any_instance_of(Net::LDAP).to receive(:auth).with('example_bind_dn',
+                                                               'example_bind_password')
+
+      # once as bind dn for connection test, once as user dn
+      expect_any_instance_of(Net::LDAP).to receive(:bind).exactly(2).times.and_return(true)
+
+      # search for user_dn
+      search_filter = Net::LDAP::Filter.eq('uid', 'bob')
+      # Net::LDAP.any_instance.expects(:search)
```
Comment (Thunderguts13) [inline spec/unit/lib/ldap_connection_spec.rb:46]: remove commented code

File: spec/unit/models/team_spec.rb [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+      private: false
+    }
+
+    team = Team.create(bob, params)
+    expect(team.valid?).to eq(false)
+  end
+
+  private
+
+  def decrypt_private_key(user)
```
Comment (Thunderguts13) [inline spec/unit/models/team_spec.rb:165]: This method also exists in the test_helper

File: spec/unit/models/teammember_spec.rb [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+describe Teammember do
+
+  let(:user) { users(:bob) }
+
+  it 'does not add second user in same team' do
+    params = {
+      user_id: users(:alice).id,
+      team_id: teams(:team1).id
+    }
+    teammember = Teammember.new(params)
```
Comment (Thunderguts13) [inline spec/unit/models/teammember_spec.rb:13]: Add contexts for new and destroy

File: spec/integration/user_provides_new_ldap_pw_spec.rb [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+  let(:bob) { users(:bob) }
+
+  context 'as bob' do
+    it 'provides new ldap password and remembers old password' do
+      ldap = double
+
+      expect(GeoIp).to receive(:activated?).at_least(:once).and_return(false)
+
+      # Prepare for do
+      bob.update(auth: 'ldap')
```
Comment (psunix) [inline spec/integration/user_provides_new_ldap_pw_spec.rb:29]: update!

File: spec/policies/account_policy_spec.rb (hunk 4 above)
Comment (psunix) [reply spec/policies/account_policy_spec.rb:15]: what about something like:
let(:policy) { AccountPolicy.new(bob, account2) }

expect(policy.show?).to eq(true)

standard for all policy specs ?

Comment (psunix) [reply spec/policies/account_policy_spec.rb:15]: assert_permit is more minitest style.
but if to extensive to change .. create a new story.

File: spec/policies/recryptrequest_policy_spec.rb [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    it 'can trigger a recrypt request' do
+      assert_permit admin, Recryptrequest, :recrypt?
+      assert_permit bob, Recryptrequest, :recrypt?
+      assert_permit alice, Recryptrequest, :recrypt?
+    end
+  end
+
+  private
+
+  def ldap_user
```
Comment (psunix) [inline spec/policies/recryptrequest_policy_spec.rb:63]: let(:ldap_user) do ....

File: spec/policies/team_policy_spec.rb [hunk 12]
```diff
@@ -18,131 +24,113 @@ class TeamPolicyTest < PolicyTest
...
+      assert_permit admin, team1, :edit?
     end
-  end
-  
-  context 'admin teams' do
-    test 'admin can list all teams' do
-      assert_permit admin, Team, :index_all?
+
+    it 'cannot edit a private team' do
+      assert private_team.private?
```
Comment (psunix) [inline spec/policies/team_policy_spec.rb:78]: expect(private_team).to be_private

File: spec/policies/user/api_policy_spec.rb [hunk 13]
```diff
@@ -76,5 +81,4 @@ class User::ApiPolicyTest < PolicyTest
   def create_api_user
```
Comment (psunix) [inline spec/policies/user/api_policy_spec.rb:81]: let(:api_user) ...

Comment (psunix) [reply spec/policies/user/api_policy_spec.rb:81]: there's also let! if required ....

File: spec/unit/models/item_spec.rb [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+      account_id: accounts(:account2).id,
+      file: items(:item1).file,
+      filename: items(:item1).filename
+    }
+    item = Item.new(params)
+    expect(item).to be_valid
+  end
+
+  it 'decrypts item' do
+    bob = users(:bob)
```
Comment (psunix) [inline spec/unit/models/item_spec.rb:33]: let

File: spec/unit/utils/changelog_reader_spec.rb [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    it 'verifies every changlog entry has at least one entry' do
+      ChangelogReader.changelog.each do |changelog_entry|
+        expect(changelog_entry.log_entries).to_not be_nil
+      end
+    end
+  end
+
+  context 'test ChangelogReader' do
+
+    subject { ChangelogReader.new }
```
Comment (psunix) [inline spec/unit/utils/changelog_reader_spec.rb:28]: check right usage of subject in rspec ... 
maybe it's better to just create let(:changelog_reader) .... in this case

File: spec/policies/account_policy_spec.rb (hunk 4 above)
Comment (TheWalkingLeek) [reply spec/policies/account_policy_spec.rb:15]: Created a separate story for this as it requires a quite amount of refactoring...

**Pull request merged**